src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,544 @@
+						givenEntry.count += 1;
+						entryFound = true;
+					}
+				}
+				if (!entryFound) {
+					times.push(newEntry);
+				}
+			}
+
+			// Assumption: After 2 entries it is recurring
+			const filteredTimes = times.filter(entry => entry.count >= 2);
+
+			for (let timeEntry of filteredTimes) {
+				const newEntry = {
+					teacher: timeEntry.teacher,
+					class: klass.longName, // note uppercase N here!
+					times: [ {
+						weekday: timeEntry.weekday,
+						startTime: timeEntry.startTime,
+						duration: timeEntry.duration,
+						room: timeEntry.room,
+					} ],
+					subject: timeEntry.subject,
+					state: 'new',
+				};
+
+				let entryFound = false;
+				for (const givenEntry of result) {
+					if (givenEntry.teacher === newEntry.teacher
+						&& givenEntry.class === newEntry.class
+						&& givenEntry.subject === newEntry.subject) {
+						givenEntry.times.push(newEntry.times[0]);
+						entryFound = true;
+					}
+				}
+				if (!entryFound) {
+					result.push(newEntry);
+				}
+			}
+		}
+
+		/**
+		 * Result: Array of {
+		 *   teacher: 'Renz',
+		 *   class: '2a',
+		 *   subject: 'mathe',
+		 *   times: [ { weekDay, startTime, endTime, room } ],
+		 *   state: 'new',
+		 * }
+		 */
+
+		return result;
+	}
+
+	/**
+	* 
+	* @param {*} session 
+	* @param {*} metaData 
+	*/
+	async fetchData(session, metaData) {
+		// Not required as all information are fetched during phase 1
+
+		return metaData;
+	}
+
+	/**
+	* 
+	* @param {*} session 
+	*/
+	async logout(session) {
+		await super.logout();
+	}
+
+	/**
+	* 
+	* @param {*} metadata 
+	*/
+	async storeMetadata(metaData, params) {
+		const webUntisMetadataService = this.app.service('webuntisMetadata');
+
+		await Promise.all(metaData.map(async (entry) => {
+			const metadataResults = await webUntisMetadataService.find({ query: {
+				datasourceId: params.datasourceId,
+				teacher: entry.teacher,
+				class: entry.class,
+				subject: entry.subject
+			}, paginate: false });
+			const result = metadataResults[0];
+
+			if (result !== undefined) { // patch/replace existing
+				await webUntisMetadataService.patch(result._id, { state: entry.state, times: entry.times });
+			} else { // create new
+				await webUntisMetadataService.create(Object.assign(
+					{ datasourceId: params.datasourceId },
+					entry
+				));
+			}
+		}));
+	}
+
+	/**
+	* 
+	* @param {*} data 
+	*/
+	async createCourses(config, data, params) {
+		// Convert metadata to actual schulcloud courses
+		// And reflect changes in metadata store
+		const webUntisMetadataService = this.app.service('webuntisMetadata');
+
+		const importCondition = this.getImportConditionEvaluator(params);
+
+		for (let entry of data) {
+			if (importCondition(entry._id.toString())) {
+				await this.obtainAndUpdateCourseAndClass(config, entry)
+					.then(() => webUntisMetadataService.patch(entry._id, { state: 'imported' }))
+					// .catch(() => webUntisMetadataService.patch(entry._id, { state: 'errored' }));
+					;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected this semicolon to be at the end of the previous line.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)